### PR TITLE
consolidate trace_printf and trace_printf_str into trace_printf

### PR DESCRIPTION
**Pull request by jjr840430**

### CODE REVIEW

File: common.c [hunk 1]
```diff
@@ -65,6 +65,88 @@ int g_enable_tracing = 1;
 struct descriptor_info **g_descriptor_list;
 unsigned int g_descriptor_list_size;
 
+// special strings to be displayed with ANSI colors
```
Comment (efiop) [inline common.c]: Please don't use c++-style comments. Use /* Your comment here */ instead.

File: common.c [hunk 2]
```diff
@@ -65,6 +65,88 @@ int g_enable_tracing = 1;
...
+	{ '\0', NULL }
+};
+
+static void
+trace_printf_str(const char *str)
+{
+	char *str_print;
+
+	// get original function pointers
+	rtr_malloc_t real_malloc = RETRACE_GET_REAL(malloc);
```
Comment (efiop) [inline common.c]: Please don't mix declarations and function calls. I.e. it should look something like:

char *str_print;
rtr_malloc_t real_malloc;
rtr_free_t real_free;
....

real_malloc = RETRACE_GET_REAL(malloc);
real_free = RETRACE_GET_REAL(free);

File: common.c (hunk 1 above)
Comment (jjr840430) [inline common.c]: Thanks for your advise.

**Comment by efiop** [comment]
Hi @jjr840430 !

Travis found some issues https://travis-ci.org/riboseinc/retrace/jobs/242758802 with your patch. You could test it out locally by running make test to make sure it works for you without errors, most of them are currently about unused variables and similar stuff.

**Comment by jjr840430** [comment]
@efiop Thanks, I'm now checking.

**Comment by jjr840430** [comment]
@efiop Can you check why CI checks has failed? It seems there is no issue from compilation.

**Comment by efiop** [comment]
@jjr840430 It's style check that caught an issue:

WARNING: space prohibited between function name and open parenthesis '('
#50: FILE: test/str.c:35:
+	char *p = strstr (s, "[test\n]");

Please fix it.

**Comment by efiop** [comment]
@jjr840430 Perfect! Thank you.

**Comment by riataman** [comment]
1) This is disabling useful tests in  test/str.c .

2) trace_printf_str is calling strlen directly rathern than real_strlen in one spot.

3) This is removing the [SNIP] functionality which was the entire point of the trace_printf_str function!

**Comment by jjr840430** [comment]
@erikbor I fixed that issue, Can you review and merge?

File: test/str.c [hunk 3]
```diff
@@ -27,11 +27,13 @@
...
 {
-	char *s = "This is a testing string. All work and no play makes Jack a dull boy. Lorem ipsum dolor sit amet, consectetur adipiscing elit";
+	// char *s = "This is a testing string. All work and no play makes Jack a dull boy. Lorem ipsum dolor sit amet, consectetur adipiscing elit";
+	char *s = "This is a testing";
 	char buf[1024];
 	int a = 2;
 
-	char *p = strstr (s, "dull");
+	char *p = strstr(s, "[test\n]");
+#if 0
```
Comment (riataman) [inline test/str.c]: Why are we removing this tests?

File: common.c [hunk 4]
```diff
@@ -85,66 +178,17 @@ trace_printf(int hdr, const char *fmt, ...)
...
 	set_tracing_enabled(old_tracing_enabled);
-}
-
-void
-trace_printf_str(const char *string)
-{
-	static const char CR[] = VAR "\\r" RST;
-	static const char LF[] = VAR "\\n" RST;
-	static const char TAB[] = VAR "\\t" RST;
-	static const char SNIP[] = "[SNIP]";
```
Comment (riataman) [inline common.c]: The whole point of this function was the [SNIP] stuff, which you didn't carry over to your rewrite.

File: common.c [hunk 5]
```diff
@@ -65,13 +65,106 @@ int g_enable_tracing = 1;
...
+		int idx = 0;
+
+		/* check NULL */
+		if (!colors->color_str)
+			break;
+
+		while (*p != '\0') {
+			if (*p == colors->ch) {
+				char *tmp;
+				size_t size = strlen(str_print) + strlen(colors->color_str) + 1;
```
Comment (riataman) [inline common.c]: We need to be careful to call our replacements directly as we might fall into loops. Call real_strlen instead.

File: common.c (hunk 4 above)
Comment (jjr840430) [inline common.c]: @riataman Can you advise what's mean of SNIP?

**Comment by jjr840430** [comment]
@drystone The goal of this issue is to consolidate trace_printf() and trace_printf_str().
trace_printf_str() function has only function to display special characters by some colors.
I think it may be consolidated into single trace_printf() function.

**Comment by jjr840430** [comment]
@riataman 
I think [SNIP] is to show whenever logging string has been truncated by limitation of the buffer size in old trace_printf_str() function.
But current trace_printf_str() function allocates buffers by dynamically if it needs.
So I think we don't need that in this issue.

**Comment by drystone** [comment]
@jjr840430 that's correct re SNIP! I'm not sure how you'd consolidate the two without losing functionality. `trace_printf_str` replaces `"\n"` with `"\\n"` which wouldn't be desirable in `trace_printf` unless you wanted never to print a newline :)

**Comment by jjr840430** [comment]
@drystone Can you review that function in this branch?
If you have any opinions, then please let me know.

**Comment by jjr840430** [comment]
@riataman I have new updates. Can you please review the changes?

**Comment by jjr840430** [comment]
@drystone Thanks for your advise.
@erikbor Can you clarify again about consolidation of two functions?

**Pull request closed without merging**